sound/soc/sof/intel/hda-loader.c

```diff
@@ -177,13 +176,6 @@ static int cl_dsp_init(struct snd_sof_dev *sdev, int stream_tag)
 			__func__);
 
 err:
-	flags = SOF_DBG_DUMP_REGS | SOF_DBG_DUMP_PCI | SOF_DBG_DUMP_MBOX;
-
-	/* force error log level after max boot attempts */
-	if (hda->boot_iteration == HDA_FW_BOOT_ATTEMPTS)
-		flags |= SOF_DBG_DUMP_FORCE_ERR_LEVEL;
-
-	hda_dsp_dump(sdev, flags);
 	snd_sof_dsp_core_power_down(sdev, chip->host_managed_cores_mask);
```

Comment:
I think this change the behaviour a bit. The intent here is to print the registers for each attempt, but only bump trace level to error on last one. So with your patch, I think the hda_dsp_dump should be ignored last time as it's printed in the above layer.

Reply:
My intention is to remove the hda_dsp_dump() calls from platform code and have it only in the core. This way it can be managed in one place.
Before this patch the dump was done for all failed try but with dev_dbg().
I'm not sure how frequent was this, but given that we have the code I would say that it was 'normal' to have failed boots of the DSP?

With the change we will print the dump once, after we run out of retry count - so in case of final error.

Reply:
@ujfalusi I'm fine with this. The original/current code explicitly dumped registers for all attempts, but I don't think we have any case where the dumps would have been different (between 1st... to Nth attempt).

Reply:
I think this is the only thing that I'm not sure about here. In general I'd think, that if all attempts failed, the most useful information is probably in the first failure, right? I don't remember ever seeing such cases with different data, but maybe it would be better to cache the first failure in such cases and then print it if all attempts failed? Also we do print at least a note if some boot attempts failed but we recovered? For debugging it might be good to also have dumps from such recovered failures, but that would needlessly alarm "normal" users, I guess. So, I wouldn't NAK this PR because of this, if everybody else thinks this is ok, but maybe this is something to consider.

Reply:
@lyakh, I understand your concern, let me explain my reasoning on  how the series ended up like this.
the aim is to reduce the noise from debugging without benefit and at the same time to try to get a level of consistency between platforms and how they report firmware crashes. With this in mind I have decided to move (I think) all such a prints to the core out from platform drivers.
To answer directly your points:
- print number of attempts if it is not 1 when FW boots: I thought about it, but decided to drop that since for the user it is not going to give too much information that everything is OK, but it took a bit more effort to get there and for me it looks like that it is expected that we need to nag the FW to boot for unknown reasons. If we print, then it should be in dev_dbg(), I think
- If we want to print the first failed boot's information then I would move the boot retry code from the platform drivers to core and do it there also to - again - have consistent logs among platforms. This is a bit big work as hda does it in a different way compared to 'intel' and iMX is not doing it at all. We could create a feature request/issue/bug to make sure that it is done if the more experienced SOF developers find this feature a must have?